solr/solrj/src/test/org/apache/solr/client/solrj/impl/Http2SolrClientTest.java

```diff
@@ -542,7 +557,32 @@ public void testUpdate() throws Exception {
   }
 
   @Test
-  public void testRedirect() throws Exception {
+  public void testFollowRedirect() throws Exception {
+    final String clientUrl = jetty.getBaseUrl().toString() + "/redirect/foo";
+    try (Http2SolrClient client =
+        new Http2SolrClient.Builder(clientUrl).withFollowRedirects(true).build()) {
+      SolrQuery q = new SolrQuery("*:*");
+      client.query(q);
+    }
+  }
+
+  @Test
+  public void testDoNotFollowRedirect() throws Exception {
+    final String clientUrl = jetty.getBaseUrl().toString() + "/redirect/foo";
+    try (Http2SolrClient client =
+        new Http2SolrClient.Builder(clientUrl).withFollowRedirects(false).build()) {
+      SolrQuery q = new SolrQuery("*:*");
+      try {
+        client.query(q);
+        fail("Should have thrown an exception.");
```

Comment:
Better to use `assertThrows` if possible.

Reply:
Done!